NitroxModel/Tcp/Connection.cs

```diff
@@ -17,11 +20,34 @@ public Connection(Socket socket)
             this.Socket = socket;
             this.MessageBuffer = new MessageBuffer();
             this.Open = true;
-        }           
+        }
+
+        public void Connect(IPEndPoint remoteEP)
+        {
+            try
+            {
+                Socket.Connect(remoteEP);
+                if (!Socket.Connected)
+                {
+                    Open = false;
+                }
+            }
+            catch (SocketException)
+            {
+                Open = false;
+            }
+        }
         
         public void BeginReceive(AsyncCallback callback)
         {
-            Socket.BeginReceive(MessageBuffer.ReceivingBuffer, 0, MessageBuffer.RECEIVING_BUFFER_SIZE, 0, callback, this);
+            try
+            {
+                Socket.BeginReceive(MessageBuffer.ReceivingBuffer, 0, MessageBuffer.RECEIVING_BUFFER_SIZE, 0, callback, this);
+            } catch (SocketException)
```

Comment:
minor: watch style, catch on new line.  Also it may be worth logging the exception name.